src/Microsoft.Diagnostics.Runtime/src/DataReaders/Live/LiveDataReader.cs

```diff
@@ -26,10 +26,12 @@ internal unsafe class LiveDataReader : IDataReader
 
         public LiveDataReader(int pid, bool createSnapshot)
         {
+            // Throws
+            using Process process = Process.GetProcessById(pid);
```

Comment:
I don't think this is a good solution here.

This works 99.9% of the time, but it introduces a race condition if the process dies between the creation of this object and when we call PssCaptureSnapshot.  As a minor point it also requires us to go through whatever machinery that `Process.GetProcessById` requires and to collect the object.  That issue has to be fixed by properly checking HRESULTs/error codes and throwing a more appropriate exception.

Reply:
Did you mean `ArgumentException` isn't appropriate?

Reply:
Sorry that was unclear.  Argument exception is good, I was only worried about using GetProcessById here.